### DISCUSSION TITLE
Create a way for scores to be unsubmittable for tournament settings

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#17121] [Ideas]
Prompted from today's community meeting on score submission.
Currently, ScoreV2 doesn't submit to osu!web. Are there plans to implement a similar system somewhere in lazer?
Offline playing is possible, but means that core services such as supporter/online features won't be available while practicing for tournaments/not submitting scores.

**Comment by peppy** [comment]
What's the reason for requiring this?

**Reply by ReiFan49** [reply]
During rank restricted tournaments, some of the extended rules of those tournaments prohibiting players from gaining rank too far. Also, an extensive practice of let's say 5 weeks worth of pool (assuming they managed to survive that long) potentially affects this as well.

**Reply by peppy** [reply]
I don't get it. Why isn't the rank just taken from the point of registration? Please try and provide a full explanation with all ratifications so we can actually attempt to understand the reasoning behind this.

**Reply by Kasumisama** [reply]
Giving an answer that I hope makes enough sense as a tournament host, and I'm also not sure of how much information players will be able to see about scores in the future.

So, regarding the "letting players not gain too much ranks", it's mostly a rule I've seen enforced in most rank-restricted tournaments, because while we do take people's rank at the time of registration, having someone farm up to let's say 3 digits and still playing in a 5 digits tournament wouldn't be the best of looks to most people. I personally don't care for that rule too much, as I do believe it's what has helped nurture the "deranking" community over the past years, with players then using as much ways as possible to avoid ranking up.

A more pressing matter though, is if every single score submitted while practicing for tournaments will be viewable from leaderboards or the player's profile. This is a something that most tournament players would definitely take an issue with, which was made apparent especially at the time where V2 scores were made submittable, until OWC 2020 I believe? 

Having to basically give free access to your scores by having all of them submit on your profile to your opponents is a huge problem to most tournament players, as that puts all of your weaknesses and strong points on full view.

This argument alone should warrant a mode to not submit scores, just like V2 is currently in stable.

I hope I explained it well enough, and I've probably missed other points that other tournament staff members could have.

**Reply by smoogipoo** [reply]
> because while we do take people's rank at the time of registration, having someone farm up to let's say 3 digits and still playing in a 5 digits tournament wouldn't be the best of looks to most people

This is not something you're going to be able to prevent. Unless your solution to this is to prevent the player from being able to gain ranks _at all_ as long as they're signed up to a tournament, even via solo-play.
As long as they're currently playing in the tournament, why does their rank matter at all anymore? What is the solution here, to DQ them weeks into a tournament? Sounds pretty bad and honestly against the [rules for official tournament support](https://osu.ppy.sh/wiki/en/Tournaments/Official_support) (at least if done without the tournament committee's approval):
> If a user satisfies the sign-up criteria (if any), the tournament must not prevent those who pass the screening from participating without both ample evidence presented publicly against them and the approval of the account support team.
>
>   This includes preventing users who are perceived to be "sandbagging" from play. Should an organiser have valid concerns about the presence of such players affecting the competitive integrity of their tournament, they may raise the issue to the [Tournament Committee](https://osu.ppy.sh/wiki/en/People/Tournament_Committee) for a case-by-case review using the [tournament reports form](https://pif.ephemeral.ink/tournament-reports).

It might be interesting to add a "private" mode for multiplayer where scores are submitted but don't contribute to PP/leaderboards and aren't shown on profiles, unless the user chooses to publicise/count them in. This would of course be on top of MP link being privated under the same setting as per osu-stable.

**Reply by Kasumisama** [reply]
> This is not something you're going to be able to prevent.

.. I did not claim to that? I simply tried to explain why a tournament host may find issues with that. I'm not even defending this rule, I don't know why this answer is so defensive.

> As long as they're currently playing in the tournament, why does their rank matter at all anymore? What is the solution here, to DQ them weeks into a tournament? Sounds pretty bad

I geniunely don't see how tournament hosts expecting players to not be out of the rank range in a, well, rank-restricted tournament is something that bad? 
It sounds like it's absolute rigging if we listened to you.

Once again, I did not defend this rule, I even said myself it's more of an issue in rank-restricted tournaments than anything. But letting people rank up to whatever rank they want, and forcing people to not rank up for 2 months are both really bad in my opinion.

Thank you for the reminder of the official tournament support rules, I definitely needed them as an active tournament host and never knew about them in my entire life. I'd appreciate it if I could not get mocked pretty blatantly when trying to be helpful. This is probably why most people don't want to come on GitHub to give feedback, lol. 

Especially when I've said the main reason for this being needed for tournaments is NOT the rank of players, which is barely relevant here, but the fact that practice scores could be public.

The "private" mode for multiplayer would be good, and the best course of action over this.

**Reply by smoogipoo** [reply]
> I geniunely don't see how tournament hosts expecting players to not be out of the rank range in a, well, rank-restricted tournament is something that bad?

Read the first part of what you quoted: "as long as they're currently playing in the tournament". If you disagree with this, then please answer the latter part of what you quoted, because you do seem to be defending it.

Just for the record, I don't think we disagree here. I'm saying that during a tournament, as in while a tournament is progressing, you can't just DQ someone because of whatever tournament rule. You need to go through the tournament committee to do that.

My point is that that's not something that the game can or will handle automatically because it effectively means that you need to prevent people from gaining PP during solo play for the duration of the tournament. I don't see another way for the game to handle this, so it may as well just not be a point of conversation.

But all of this is irrelevant as you said, because the topic is private multiplayer scores. I'm not sure why rank ranges were brought up in the first place, and I'm just trying to make the point that the game can't really do anything about this.

Not sure why you think I'm mocking you, I'm just trying to explain and clarify things so irrelevant points can stop being discussed.

**Reply by bdach** [reply]
@kasumiikun please, there's no need for the inflammatory tone or immediately assuming malice in every statement. it doesn't help anything.

**Reply by Cookie-Tree** [reply]
Going back to the original question, I would like to repeat this part of kasumii's reply which is core to the thread:

> A more pressing matter though, is if every single score submitted while practicing for tournaments will be viewable from leaderboards or the player's profile. This is a something that most tournament players would definitely take an issue with, which was made apparent especially at the time where V2 scores were made submittable, until OWC 2020 I believe?

This was changed during OWC 2020 as tournament players preferred to be able to play a map without having to make a separate unsubmitted difficulty nor having to log out. This is a massive inconvenience and can easily leak scores to opposing teams if players are not careful, especially if they are not aware of scorev2 submitting to web again.

This thread got out of hand quick which is why I want to underline that this is still a much desired feature for tourney players, regardless of what kind of tourney it is.

**Reply by peppy** [reply]
For features already implemented (in stable), you can be assured that they will be added. The reason this hasn't been added yet is mostly https://github.com/ppy/osu/discussions/17121#discussioncomment-2305760

**Reply by Cookie-Tree** [reply]
I should have clarified that I was mainly talking about this feature for singleplayer instead of multiplayer, as that's the issue @zachmanthethird intended to raise. Or is that also what you meant with this?
> For features already implemented (in stable), you can be assured that they will be added.

**Reply by Akarinnnnn** [reply]
> I should have clarified that I was mainly talking about this feature for singleplayer instead of multiplayer, as that's the issue @zachmanthethird intended to raise. Or is that also what you meant with this?
> 
> > For features already implemented (in stable), you can be assured that they will be added.

Note that lazer accepts playing solo in a multiplayer room, creating a practice room for yourself to disable submission is possible.

**Reply by bdach** [reply]
@Akarinnnnn I feel like there is some confusion here because of you bumping a 4-year-old thread. playing in a multiplayer room *does cause a score submission* now. regardless of whether you're alone in the room or not.

**Reply by Akarinnnnn** [reply]
> playing in a multiplayer room _does cause a score submission_ now. regardless of whether you're alone in the room or not.

First I apologize to bumping this out. In context of this thread, as a tournament player, I want to "not submit" my scores, is majorly because of leaking performance too early is a risk, which will make opponents get advantages on ban/pick. Considering prevent submitting breaks lazer score submission constraint, instead of avoiding scores being submitted, we can simply hide them, as peppy said:
> Either that or room setting that turns off all score submission.

To remain ranking system's integrity, scores can be revealed in my imagination, I prefer to reveal them after whole tournament completed.

> playing in a multiplayer room does cause a score submission now.

As a regular lazer player, I noticed multiplayer room submits scores on this feature online. In this context, I want to talk about a room level toggle, which can hide players' scores. More ever, a tourney code input which can be used to associate this room to a specific lazer tournament, allows reveal all scores later in the room.

**Comment by peppy** [comment]
Importantly: this is describing a proposal for a future. multiplayer scores currently do not "submit" due to actual oversights in the way lazer score submission is structured. Will be addressed at some point. (although i believe there's also no way to hide multiplayer history in the newer systems, so we may need to look at addressing that as well).

**Comment by zachmanthethird** [comment]
Is this being considered in either the `Usable for tournaments` or `Multiplayer and playlists` milestones?
From #18775, you noted that issues in milestones should be "used by the majority of players (50% or more)" but this would affect 100% of tournament players, even if that is not 50% or more of regular players.

**Reply by smoogipoo** [reply]
This isn't being considered for those - those milestones are only what's required for a first "release". I don't think we've finished discussing this internally yet, I believe last time we ended up on discussing some way to private scores (similar to the way multiplayer matches can be privated) but not awarding PP until they're unprivated.

**Reply by peppy** [reply]
Either that or room setting that turns off all score submission.
